ui/src/pages/AddToTrip.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import Storage from "storage";
+
+import { GetTrips, GetTrip, UpdateTrip } from 'reducers/trips';
+
+import Title from 'components/text/Title';
+import Trip from 'models/trip';
+import * as Image from 'models/image';
+import Section from 'components/section/index';
+import Button from 'components/button/Button';
+import TripTile from  'components/tile/Trip';
+
+import requireAuth from './requireAuth';
+
+class AddToTrip extends React.Component{
+    componentWillMount() {
+        this.props.getTrips();
+      }
+    
+    getUpcomingTitle = () => {
+        if (this.props.gettingTrips)
+          return "Fetching Trips...";
+        if (this.props.trips.length === 0)
+          return "";
+        return "Your Trips";
+    }
+
+    updateTrip = (trip) => {
+        var event = this.props.event;
+        const newTrip = Trip.fromObject(trip.toObject());
+        newTrip.events.push(event);
+        if (newTrip.background === Image.getBlackImage()) newTrip.background = event.images[0];
+        this.props.updateTrip(trip.id, newTrip);
+    }
+
+    handleClick = (id) =>{
+        this.props.getTrip(id, trip => this.updateTrip(trip));
```

Comment:
What is this trying to do here?

Reply:
Oh got it! I was confused because I thought it was just updating the trip with itself